tests/codim_one/integrate_one_over_r_telles.cc

```diff
@@ -0,0 +1,302 @@
+// ---------------------------------------------------------------------
+// $Id: integrate_one_over_r.cc 30338 2013-08-18 22:02:27Z heltai $
+//
+// Copyright (C) 2005 - 2013 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+// integrates the function *f(x,y)/R, where f(x,y) is a power of x and
+// y on the set [0,1]x[0,1]. dim = 2 only.
+
+#include <fstream>
+#include <deal.II/base/logstream.h>
+#include <deal.II/base/utilities.h>
+
+// all include files needed for the program
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/base/geometry_info.h>
+#include <deal.II/fe/fe_q.h>
+
+#include <fstream>
+#include <string>
+
+#include <math.h>
+
+using namespace std;
+using namespace dealii;
+
+// We test the integration of singular kernels with a singularity of kind 1/R
+// We multiply this function with a polynomial up to degree 6.
+
+double
+exact_integral_one_over_r (
+    const unsigned int i, const unsigned int j,
+    const unsigned int vertex_index);
+
+ofstream logfile("output");
+
+int
+main ()
+{
+  deallog.attach(logfile);
+  deallog.depth_console(0);
+  deallog << std::fixed;
+
+  deallog << endl
+          << "Calculation of the integral of f(x,y)*1/R on [0,1]x[0,1]" << endl
+          << "for f(x,y) = x^i y^j, with i,j ranging from 0 to 5, and R being" << endl
+          << "the distance from (x,y) to four vertices of the square." << endl
+          << endl;
+
+
+  std::vector<Point<2> > vertices = FE_Q<2>(1).get_unit_support_points();
+
+  for (unsigned int m=1; m<7; ++m)
+    {
+      deallog << " =========Quadrature Order: " << m
+              << " =============================== " << endl;
+      deallog << " ============================================================ " << endl;
+      unsigned int index=0;
+        {
+          deallog << " ===============Vertex: " << vertices[index]
+                  << " ============================= " << endl;
+          QTelles<2> quad(6, vertices[index]);
+          QGaussOneOverR<2> quad2(4, vertices[index]);
```

Comment:
The quadrature order is never used. You are always checking Telles of order 6, and OneOverR of order 4. Can you put m in here?